sound/soc/sof/topology.c

```diff
@@ -809,6 +809,68 @@ static int sof_parse_tokens(struct snd_soc_component *scomp,  void *object,
 				    array_size, 1, 0);
 }
 
+static int sof_w_no_wname_in_long_name(struct snd_soc_component *scomp,
+				       struct snd_soc_dapm_widget *w, int type,
+				       struct snd_soc_tplg_vendor_value_elem *v)
+{
+	if (!(type == SND_SOC_TPLG_TUPLE_TYPE_WORD ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_SHORT ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_BYTE ||
+	      type == SND_SOC_TPLG_TUPLE_TYPE_BOOL)) {
+		dev_err(scomp->dev, "Bad tuple type %d\n", type);
+		return -EINVAL;
+	}
+
+	w->no_wname_in_kcontrol_name = le32_to_cpu(v->token);
+
+	return 0;
+}
+
+static int sof_dapm_widget_init(struct snd_soc_component *scomp,
+				struct snd_soc_dapm_widget *w,
+				struct snd_soc_tplg_vendor_array *array,
+				int array_size)
+{
+	while (array_size > 0) {
+		int asize, type, i;
+
+		asize = le32_to_cpu(array->size);
+
+		/* validate asize */
+		if (asize <= 0) {
+			dev_err(scomp->dev, "error: invalid array size 0x%x\n",
+				asize);
+			return -EINVAL;
+		}
+
+		/* make sure there is enough data before parsing */
+		array_size -= asize;
+		if (array_size < 0) {
+			dev_err(scomp->dev, "error: invalid array size 0x%x\n",
+				asize);
+			return -EINVAL;
+		}
+		type = le32_to_cpu(array->type);
+
+		for (i = 0; i < le32_to_cpu(array->num_elems); i++) {
+			int ret = 0;
+
+			switch (le32_to_cpu(array->value[i].token)) {
+			case SOF_TKN_COMP_NO_WNAME_IN_KCONTROL_NAME:
+				ret = sof_w_no_wname_in_long_name(scomp, w, type,
+								  &array->value[i]);
+				break;
+			}
```

Comment:
do we really need a switch case here, why not use a regular if()?

and if you do want a switch case, you need a default: case

Reply:
I am just thinking if in the future there are other values in the topology that should be copied over to dapm_widget. I can add the default, but it will be empty.